ci/upload-bazel-metrics.yml

```diff
@@ -0,0 +1,54 @@
+# Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+steps:
+  - template: bash-lib.yml
+    parameters:
+      var_name: bash_lib
+  - bash: |
+      set -euo pipefail
+      if [[ $os = windows ]]; then
+      else
+        eval "$(dev-env/bin/dade assist)"
+        source $(bash_lib)
+      fi
+      jq -nc >job-md.json '{
+        timestamp: "$(pipeline_timestamp)",
+        id: "$(pipeline_id)",
+        agent_id: "$(Agent.Id)",
+        agent_job_name: "$(Agent.JobName)",
+        agent_machine_name: "$(Agent.MachineName)",
+        agent_name: "$(Agent.Name)",
+        agent_os: "$(Agent.OS)",
+        agent_os_architecture: "$(Agent.OSArchitecture)",
+        build_build_id: "$(Build.BuildId)",
+        build_build_number: "$(Build.BuildNumber)",
+        build_definition_name: "$(Build.DefinitionName)",
+        build_source_branch: "$(Build.SourceBranch)",
+        build_source_branch_name: "$(Build.SourceBranchName)",
+        build_source_version: "$(Build.SourceVersion)",
+        system_job_attempt: "$(System.JobAttempt)",
+        system_job_display_name: "$(System.JobDisplayName)",
+        system_job_id: "$(System.JobId)",
+        system_job_name: "$(System.JobName)",
+        system_pullRequest_pullRequestId: "$(system.pullRequest.pullRequestId)",
+        system_pullRequest_pullRequestNumber: "$(system.pullRequest.pullRequestNumber)",
+        system_pullRequest_mergedAt: "$(system.pullRequest.mergedAt)",
+        system_pullRequest_sourceBranch: "$(system.pullRequest.sourceBranch)",
+        system_pullRequest_targetBranch: "$(system.pullRequest.targetBranch)",
+        system_pullRequest_sourceRepositoryUri: "$(system.pullRequest.sourceRepositoryUri)",
+        system_pullRequest_sourceCommitId: "$(system.pullRequest.sourceCommitId)",
```

Comment:
I'm not sure what's the best approach here, but I should point out this will not work as expected once merged into main. Specifically, the `$()` syntax evaluation at the Azure level only happens if the variables are defined, and the `System.PullRequest.*` ones are not defined on runs that are not PRs (like main runs, or the manual run I just did on this branch). This results in the following JSON object:
```json
{
  "timestamp": "2021-06-07T09:20:08.0000000",
  "id": "2021-06-07T09:20:08.0000000-0da5d1d9-276d-5173-c4c4-9d4d4ed14fdb-1",
  "agent_id": "17473",
  "agent_job_name": "Linux",
  "agent_machine_name": "vsts-agent-ubuntu-20-04-gl00",
  "agent_name": "vsts-agent-ubuntu-20-04-gl00",
  "agent_os": "Linux",
  "agent_os_architecture": "X64",
  "build_build_id": "79607",
  "build_build_number": "20210607.4",
  "build_definition_name": "PRs",
  "build_source_branch": "refs/heads/ci-monitor",
  "build_source_branch_name": "ci-monitor",
  "build_source_version": "8478803b5c3fe6832d8e7c1aa2d16c217866ea98",
  "system_job_attempt": "1",
  "system_job_display_name": "Linux",
  "system_job_id": "0da5d1d9-276d-5173-c4c4-9d4d4ed14fdb",
  "system_job_name": "__default",
  "system_pullRequest_pullRequestId": "$(system.pullRequest.pullRequestId)",
  "system_pullRequest_pullRequestNumber": "$(system.pullRequest.pullRequestNumber)",
  "system_pullRequest_mergedAt": "$(system.pullRequest.mergedAt)",
  "system_pullRequest_sourceBranch": "$(system.pullRequest.sourceBranch)",
  "system_pullRequest_targetBranch": "$(system.pullRequest.targetBranch)",
  "system_pullRequest_sourceRepositoryUri": "$(system.pullRequest.sourceRepositoryUri)",
  "system_pullRequest_sourceCommitId": "$(system.pullRequest.sourceCommitId)",
  "git_branch": "",
  "git_commit": "8478803b5c3fe6832d8e7c1aa2d16c217866ea98",
  "git_tree": "791117d6ca03ae798e72f52721fc8eb35623bf13"
}
```

I _think_ the (non-secret) [Azure variables _are_ available as env vars by default](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#environment-variables), using a simple all-caps, replace-dots-with-underscores naming scheme, so it's worth trying `${SYSTEM_PULLREQUEST_PULLREQUESTID:-}` etc.

Reply:
Thanks! I'll try the env-vars.